### PR TITLE
ci: add a github action to generate and publish docs

### PR DESCRIPTION
**Pull request by mythi**
Closes: #399 

/cc @intelkevinputnam

### CODE REVIEW

**Comment by bart0sh** [comment]
@mythi lgtm. Is it possible to see a result somewhere?

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up doc directory
+      run: |
+        mkdir $HOME/output
+        touch $HOME/output/.nojekyll
+    - name: Build latest
+      run: |
+        rm -rf _work/venv
+        make vhtml
+        mv _build/html/* $HOME/output/
```
Comment (rojkov) [inline .github/workflows/publish.yml:27]: I guess this move operation could be faster if it was
```
make vhtml
touch _build/html/.nojekyll
mv _build/html $HOME/output
```

Then the step "Set up doc directory" also could be removed.

**Comment by mythi** [comment]
> @mythi lgtm. Is it possible to see a result somewhere?

@bart0sh I did not enable this workflow for my branch to try but I did run the steps manually (except `git push`)

File: .github/workflows/publish.yml (hunk 1 above)
Comment (mythi) [reply .github/workflows/publish.yml:27]: I copied these from pmem-csi. In the near future, this workflow needs to support builds for the release branch(es) and master  so there will be subdirectories under `$HOME/output` and a run stage for each branch...

**Review by rojkov — APPROVED** [review]
Looks good to me

**Pull request merged**